### PR TITLE
Production deployment [Release] files addition . 

### LINKED ISSUE #295: Implement production release process
**Is your feature request related to a problem? Please describe.**
Our automated pipeline currently deploys to the staging environment at nest.owasp.dev. As the project evolves, it's time to establish a dedicated production environment for nest.owasp.org, which currently redirects to staging.

**Describe the solution you'd like**
Enhance CI/CD workflows, Docker configurations, Ansible playbooks, and any other necessary setups to enable automated deployments to the Nest production environment triggered by creating a GitHub release.

### PR DESCRIPTION
**Pull request by yashpandey06**
- Added `build-docker-production-images` and `deploy-nest-production` jobs to handle the production deployment process.
- The `build-docker-production-images` job builds and pushes production Docker images for both frontend and backend using production-specific Dockerfiles.
- Modified the condition for the production deployment to trigger on new release creation (`github.event_name == 'release' && github.event.action == 'created'`).
- Set up production-specific environment variables and Docker build configurations.

Fixes #295

### CODE REVIEW

**Comment by yashpandey06** [comment]
@arkid15r ..the changes are in progress but still would like to have your views on it .

I would like to propose an update to our branching strategy and deployment process. We can introduce a 'develop' branch, which will be the primary branch for contributors to check out and work on new features or changes. Staging deployments will be triggered from the 'develop' branch, keeping staging separate from production-ready code. Once the changes in 'develop' are finalized and tested, they will be merged into the 'main' branch, which will be solely responsible for production deployments. This  approach will help ensure that only stable, reviewed changes are deployed to production, providing an added layer of security.

Let m e know your views .

Thanks !!

File: .github/ansible/production/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+- name: Deploy Nest to Production
+  hosts: nest_production
+  tasks:
+    - name: Copy Nginx configuration file
+      ansible.builtin.copy:
+        src: '{{ github_workspace }}/nginx/staging.conf'
```
Comment (arkid15r) [inline .github/ansible/production/deploy.yml]: We want a separate config.

File: .github/workflows/ci-cd.yaml [hunk 2]
```diff
@@ -232,3 +235,81 @@ jobs:
...
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build backend image
+        uses: docker/build-push-action@v6
+        with:
+          context: backend
+          file: backend/Dockerfile.production
+          platforms: linux/amd64
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/owasp-nest-backend:latest
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yaml]:
```suggestion
          tags: ${{ secrets.DOCKERHUB_USERNAME }}/owasp-nest-backend:release
```

File: .github/workflows/ci-cd.yaml [hunk 3]
```diff
@@ -232,3 +235,81 @@ jobs:
...
+
+      - name: Prepare frontend environment
+        run: |
+          echo "VITE_ALGOLIA_APP_ID=${{ secrets.ALGOLIA_APP_ID }}" > frontend/.env.production
+          echo "VITE_ALGOLIA_SEARCH_KEY=${{ secrets.ALGOLIA_SEARCH_API_KEY }}" >> frontend/.env.production
+          echo "VITE_NEST_API_URL=https://nest.owasp.dev/api/v1" >> frontend/.env.production
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yaml]:
```suggestion
          echo "VITE_NEST_API_URL=https://nest.owasp.org/api/v1" >> frontend/.env.production
```

File: .github/workflows/ci-cd.yaml [hunk 4]
```diff
@@ -232,3 +235,81 @@ jobs:
...
+          echo "VITE_NEST_ENV=production" >> frontend/.env.production
+
+      - name: Build frontend image
+        uses: docker/build-push-action@v6
+        with:
+          context: frontend
+          file: frontend/Dockerfile.production
+          platforms: linux/amd64
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/owasp-nest-frontend:latest
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yaml]:
```suggestion
          tags: ${{ secrets.DOCKERHUB_USERNAME }}/owasp-nest-frontend:release
```

File: docker-compose-production.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
+services:
+  backend:
+    container_name: nest-backend
+    image: arkid15r/owasp-nest-backend:latest
```
Comment (arkid15r) [inline docker-compose-production.yml]:
```suggestion
    image: arkid15r/owasp-nest-backend:release
```

File: docker-compose-production.yml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    restart: unless-stopped
+    networks:
+      - nest_app_network
+      - nest_db_network
+    volumes:
+      - ./data:/home/owasp/data
+
+  frontend:
+    container_name: nest-frontend
+    image: arkid15r/owasp-nest-frontend:latest
```
Comment (arkid15r) [inline docker-compose-production.yml]:
```suggestion
    image: arkid15r/owasp-nest-frontend:release
```

**Review by arkid15r — CHANGES_REQUESTED** [review]

File: .github/ansible/production/deploy.yml (hunk 1 above)
Comment (yashpandey06) [reply .github/ansible/production/deploy.yml]: Sure ...will be creating production.conf

Comment (yashpandey06) [reply .github/ansible/production/deploy.yml]: @arkid15r I have introduced production.conf now ..let me know if all the changes are good to go...and also should i implement the branch strategy also ? or keep it for another PR ..
Thanks !

**Review by arkid15r — APPROVED** [review]
Let's try it

**Pull request merged**